scripts/o3de/o3de/manifest.py

```diff
@@ -314,6 +314,24 @@ def get_project_external_subdirectories(project_path: pathlib.Path) -> list:
                         project_object['external_subdirectories'])) if 'external_subdirectories' in project_object else []
     return []
 
+def get_project_engine(project_path: str or pathlib.Path) -> pathlib.Path or None:
+    project_object = get_project_json_data(project_path=project_path)
+    if project_object:
+        engine_name = project_object.get('engine','')
+        if engine_name:
+            engine_path = get_registered(engine_name=engine_name)
+            if engine_path:
+                return engine_path.as_posix()
+    
+    engine_paths = get_manifest_engines()
+    resolved_project_path = pathlib.Path(project_path).resolve()
+    for engine_path in engine_paths:
+        projects = get_engine_projects(engine_path)
```

Comment:
This is code returns an ambiguous engine path and I think should be removed.
A project needs to be registered the engine it is with to avoid this scenario.

It is returning the first engine which has the project registered.

What this means is if for some reason the "engine" field from AutomatedTesting `project.json` was removed, it just return the engine of the first project it finds which may be incorrect.

The query code o3de object code in the manifest is authored in a way to not return ambiguous results.

Just registering an engine again, would move it to the top of the list of "engines" in the `o3de_manifest.json` which change the result of this function.

Reply:
Ah I need to include a test for this code anyways to prove what it's doing.

The goal of this code is, given the absolute project path like `c:\o3de\AutomatedTesting` which isn't registered in the `o3de_manifest.json`, find out if it was registered in any `engine.json`, by comparing the absolute path of the project with the absolute path of every project registered with that `engine.json`.

`get_engine_projects` returns an absolute path to the projects - so if the engine is `C:/o3de` and the project is `AutomatedTesting` you get back `C:/o3de/AutomatedTesting`,  and if the engine is `D:/o3de-sdk` and the project is still `AutomatedTesting` you get back `D:/o3de-sdk/AutomatedTesting`.  In both cases, I would compare the passed in absolute project path to see which one is the correct engine.

It's really just a fallback for when a project isn't registered with an engine correctly so likely it can be removed and some kind of warning/error displayed.

Reply:
I've added tests for this code that show it is able to find the engine in the parent folder.

Reply:
It would be simpler to just scan upwards from the project path to a directory with an engine.json within it.

That would skip the need to lookup the manifest engines

```python
engine_path = utils.find_ancestor_dir_containing_file(pathlib.PurePath('engine.json'), resolved_project_path)
if engine_path:
    projects = get_engine_projects(engine_path)
    # ...
```